### PR TITLE
[CU-1ux39qh] Validation/oracle

### PR DESCRIPTION
**Pull request by ysfkel**

### CODE REVIEW

File: frame/oracle/src/benchmarking.rs [hunk 1]
```diff
@@ -41,10 +38,10 @@ benchmarks! {
 	add_asset_and_info {
 		let caller = T::AddOracle::successful_origin();
 		let asset_id = 1;
-		Validated::new(Percent::from_percent(80)).unwrap(),
-		Validated::new(3).unwrap(),
-		Validated::new(5).unwrap(),
-		Validated::<T::BlockNumber, ValidBlockInterval<T::StalePrice>>::new(5).unwrap(),
+		Validated::new(Percent::from_percent(80)).unwrap();
```
Comment (dzmitry-lahoda) [inline frame/oracle/src/benchmarking.rs:41]: in lending benches are part of tests and run as part, so these will never break

File: frame/oracle/src/lib.rs [hunk 2]
```diff
@@ -457,23 +461,27 @@ pub mod pallet {
...
 			origin: OriginFor<T>,
 			asset_id: T::AssetId,
-			threshold: Percent,
-			min_answers: u32,
-			max_answers: u32,
-			block_interval: T::BlockNumber,
+			valid_threshold: Validated<Percent, ValidThreshhold>,
+			valid_min_answers: Validated<u32, ValidMinAnswers>,
+			valid_max_answers: Validated<u32, ValidMaxAnswer<T::MaxAnswerBound>>,
+			valid_block_interval: Validated<T::BlockNumber, ValidBlockInterval<T::StalePrice>>,
```
Comment (KaiserKarel) [inline frame/oracle/src/lib.rs:467]: I don't think there is a need to name these valid_foo to be honest. The type reflects what it is.

File: frame/oracle/src/lib.rs [hunk 3]
```diff
@@ -457,23 +461,27 @@ pub mod pallet {
...
 			reward: BalanceOf<T>,
 			slash: BalanceOf<T>,
 		) -> DispatchResultWithPostInfo {
 			T::AddOracle::ensure_origin(origin)?;
-			ensure!(min_answers > 0, Error::<T>::InvalidMinAnswers);
+
+			let threshold = valid_threshold.value();
+			let min_answers = valid_min_answers.value();
+			let max_answers = valid_max_answers.value();
+			let block_interval = valid_block_interval.value();
```
Comment (KaiserKarel) [inline frame/oracle/src/lib.rs:476]: Can we make the Deref impl public to avoid the need to remap names?

Comment (dzmitry-lahoda) [reply frame/oracle/src/lib.rs:476]: deref would eventually lead to clones if we use move:
```
let (x,y,z) = (z.into(), y.into(), z.into());
```
or as in lending - if there are so much parameters (and new may be added for v2) - create struct.

Comment (dzmitry-lahoda) [reply frame/oracle/src/lib.rs:476]: they are thinking about DerefMove for several years:)

Comment (dzmitry-lahoda) [reply frame/oracle/src/lib.rs:476]: oh, in this case it is all copy values. will work.

File: frame/oracle/src/benchmarking.rs (hunk 1 above)
Comment (cor) [reply frame/oracle/src/benchmarking.rs:41]: What do you mean by this?

File: frame/oracle/src/lib.rs (hunk 2 above)
Comment (cor) [reply frame/oracle/src/lib.rs]: Fixed in 628e2577c713fb871ca3aae8c98926b0bf9383b0

File: frame/oracle/src/lib.rs (hunk 3 above)
Comment (cor) [reply frame/oracle/src/lib.rs]: fixed in 43da2998d2db1e50149a6882807743a7684de31e

**Review by cor — DISMISSED** [review]
Hmm, `cargo test -p pallet-oracle` was succesfull, but CI is failing

**Comment by KaiserKarel** [comment]
Still some Ci errors :*(

**Review by cor — APPROVED** [review]
LGTM

**Pull request merged**